doc/erofs.md

```diff
@@ -0,0 +1,431 @@
+        used.  Read about this in the inode section.
+*   `blkszbits`: log2 of the block size.  Better set this to 12 (4096).
+*   `root_nid`: the reference to the root inode.  See the inodes section for
+    what that means.  Normally inodes are stored in u64, but this is somewhat
+    randomly a u16, which means that you're gonna need to put the root
+    directory near the start.
+*   `inos`: the total number of inodes defined.  This is only used for
+    `statfs()` purposes.
+*   `build_time`, `build_time_nsec`: this is something like a compression
+    feature if you want all (or many) files in your filesystem to have the same
+    mtime.  Then you can use the "compact" inode layout, which doesn't have its
+    own `mtime` field, and this one will be used instead.  If you don't have
+    compact inodes then this is meaningless.
+*   `blocks`: total filesystem block size.  This is only used for `statfs()`.
+*   `meta_blkaddr`: the start of the "metadata area".  This is where the inodes
+    are.  This is a block address, so it gets multiplied by the block size to
+    determine the actual offset.
+*   `xattr_blkaddr`: the start of the "shared xattr area".  See the "Shared
+    xattr" and "Inodes" sections for more info.
+
+## Extended attributes
+
+There are two options for storing xattr data in a erofs:
+*   inline with the inode itself
+*   in a "shared xattr" struct somewhere
+
+The format of both of these is the same.
+
+The inline thing is nice and simple, but it might be space-inefficient for
+cases where the same (key, value) pair appears over and over again (which might
+be the case for things like security labels and acls and the like).
+
+### Prefix indexes
+
+A rudimentary form of compression is supported on xattr names.  There are a
+number of hardcoded "common prefixes" defined with the `EROFS_XATTR_INDEX_`
+constants in `erofs_fs.h`.  Confusingly, although `LUSTRE` is present, it's not
+wired up in the kernel.  Don't use that one.
+
+The basic idea is that you find the prefix for your xattr from the list (like
+`user.` or `security.`) and then you store only the "suffix" part, along with
+the prefix index.  If you can't find a prefix, you use 0 (which is conceptually
+a prefix of "").  If the prefix matches the entire name then the suffix is `""`.
+
+Note: you really need to do this "compression" step, because it's assumed
+during the lookup phase.  ie: if we're looking for an xattr `"user.xyz"` then
+we'll only consider the entries that have the prefix index for `user.` set on
+them.  If you didn't properly "compress" your xattr names, they won't be found.
+
+There's support in the erofs format for custom prefixes.  That's when the high
+bit of the prefix index is set.  These got added circa kernel version 6.4 with
+a patch series ending with `6a318ccd7e08` ("erofs: enable long extended
+attribute name prefixes") but aren't documented here because we don't use them.
```

Comment:
Anyway, I guess you could consider to use them since this feature is landed before overlayfs `data-only` layer feature, so if the new on-disk format uses `data-only` layer feature, I guess you could safely use it too.